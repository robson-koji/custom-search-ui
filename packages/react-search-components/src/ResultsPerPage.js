import PropTypes from "prop-types";
import React from "react";

function ResultsPerPage({ onChange, options, value }) {
  return (
    <div className="results-per-page">
      <label htmlFor="results-per-page">
        Show{" "}
        <select
          name="results-per-page"
          id="results-per-page"
          value={value}
          onChange={onChange}
        >
          {options.map(option => (
            <option key={option} value={option}>
              {option}
            </option>
          ))}
        </select>
      </label>
    </div>
  );
}

ResultsPerPage.propTypes = {
  onChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.number).isRequired,
  value: PropTypes.number
};

export default ResultsPerPage;
