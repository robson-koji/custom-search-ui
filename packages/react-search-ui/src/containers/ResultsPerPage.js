import PropTypes from "prop-types";
import React, { Component } from "react";
import { withSearch } from "..";
import { ResultsPerPage } from "@elastic/react-search-components";

export class ResultsPerPageContainer extends Component {
  static propTypes = {
    results: PropTypes.arrayOf(PropTypes.object).isRequired,
    resultsPerPage: PropTypes.number.isRequired,
    searchTerm: PropTypes.string.isRequired,
    setResultsPerPage: PropTypes.func.isRequired
  };

  render() {
    const {
      results,
      resultsPerPage,
      searchTerm,
      setResultsPerPage
    } = this.props;

    if (!searchTerm && results.length === 0) return null;

    return (
      <ResultsPerPage
        onChange={e => {
          setResultsPerPage(parseInt(e.currentTarget.value, 10));
        }}
        options={[20, 40, 60]}
        value={resultsPerPage}
      />
    );
  }
}

export default withSearch(ResultsPerPageContainer);
