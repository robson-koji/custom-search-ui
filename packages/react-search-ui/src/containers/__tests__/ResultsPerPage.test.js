import React from "react";
import { shallow } from "enzyme";
import { ResultsPerPageContainer } from "../ResultsPerPage";

const params = {
  results: [{}, {}],
  resultsPerPage: 20,
  searchTerm: "test",
  setResultsPerPage: jest.fn()
};

beforeEach(() => {
  params.setResultsPerPage = jest.fn();
});

it("renders correctly", () => {
  const wrapper = shallow(<ResultsPerPageContainer {...params} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders empty when it doesn't have enough data", () => {
  const wrapper = shallow(
    <ResultsPerPageContainer
      {...{
        ...params,
        searchTerm: "",
        results: []
      }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it("will call back when a selection is made in the view", () => {
  const wrapper = shallow(<ResultsPerPageContainer {...params} />);

  wrapper.find("ResultsPerPage").prop("onChange")({
    currentTarget: {
      value: 40
    }
  });

  const resultsPerPage = params.setResultsPerPage.mock.calls[0][0];
  expect(resultsPerPage).toEqual(40);
});
