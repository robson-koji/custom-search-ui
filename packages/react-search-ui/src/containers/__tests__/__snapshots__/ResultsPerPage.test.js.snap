// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ResultsPerPage
  onChange={[Function]}
  options={
    Array [
      20,
      40,
      60,
    ]
  }
  value={20}
/>
`;

exports[`renders empty when it doesn't have enough data 1`] = `""`;

exports[`supports a render prop 1`] = `
<div>
  20
</div>
`;
